sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -559,6 +559,25 @@ case class DataSource(
       }
       globPath
     }.toSeq
+
+    if (checkFilesExist) {
+      val (filtered, filteredOut) = allGlobPath.partition { path =>
```

Comment:
Nit: I'd call `filtered` as `filteredIn` to avoid ambiguity. It might also be very slightly cleaner to avoid the `!` in the expression and flip these two values.

Reply:
I fixed it.